### PR TITLE
Update version numbers for FSharp.Core

### PR DESCRIPTION
**Pull request by KevinRansom**
````
  FSharp Language Version: 6.0
  FSharp.Core assembly Version: 6.0.0.0
  FSharp.Core package Version: 6.0.0
  FCS assembly version: 41.0.0.0
  FCS package version: 41.0.1
````

````
  LanguageFeature.AdditionalTypeDirectedConversions, languageVersion60
  LanguageFeature.RelaxWhitespace2, languageVersion60
  LanguageFeature.OverloadsForCustomOperations, languageVersion60
  LanguageFeature.ExpandedMeasurables, languageVersion60
  LanguageFeature.ResumableStateMachines, languageVersion60
  LanguageFeature.StructActivePattern, languageVersion60
  LanguageFeature.PrintfBinaryFormat, languageVersion60
  LanguageFeature.IndexerNotationWithoutDot, languageVersion60
  LanguageFeature.RefCellNotationInformationals, languageVersion60
  LanguageFeature.UseBindingValueDiscard, languageVersion60
  LanguageFeature.NonVariablePatternsToRightOfAsPatterns, languageVersion60
  LanguageFeature.AttributesToRightOfModuleKeyword, languageVersion60
````

### CODE REVIEW

**Comment by vzarytovskii** [comment]
Seems only one test failing:

```log
+++ Diagnostics\async (UsingReturnInIfThenElse.fs) +++
--------------------------------------------------------
Results from hosted compiler
msg: Compiling
cmd: fsc   --warnaserror+ --test:ErrorRanges --flaterrors UsingReturnInIfThenElse.fs 
Exit code: 1
Error:     0
Microsoft (R) F# Compiler version 12.0.0.0 for F# 6.0
Copyright (c) Microsoft Corporation. All Rights Reserved.
(6,9-6,21): error FS0193: Type constraint mismatch. The type     'Async<unit>'    is not compatible with type    'Async<int>'
--------------------------------------------------------
Compiling: [fsc   --warnaserror+ --test:ErrorRanges --flaterrors UsingReturnInIfThenElse.fs ]
Microsoft (R) F# Compiler version 12.0.0.0 for F# 6.0
Copyright (c) Microsoft Corporation. All Rights Reserved.

UsingReturnInIfThenElse.fs(6,9-6,21): error FS0193: Type constraint mismatch. The type     'Async<unit>'    is not compatible with type    'Async<int>'
Compile Unexpectedly Failed: 256 
FAIL
```

**Review by brettfo — APPROVED** [review]

**Comment by KevinRansom** [comment]
@vzarytovskii , yes it has taken some thought to figure out tmagic for that.

**Comment by KevinRansom** [comment]
@vzarytovskii , @brettfo --- Assuming it passes this is ready

**Comment by dsyme** [comment]
@KevinRansom I updated tests.fs to remove the test case corresponding to the deleted test files for array-no-dot-warnings in preview

File: tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConstructorTests.fs [hunk 1]
```diff
@@ -26,6 +26,7 @@ doSomething {Record.field1=0; field2=0}
 type Person = { Name : string; Age : int; City : string }
 let x = { Name = "Isaac", Age = 21, City = "London" }
         """
+        |> withLangVersion50
```
Comment (dsyme) [inline tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConstructorTests.fs:29]: Is there a specific reason this needed langversion 5.0?

**Review by dsyme — APPROVED** [review]

File: tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConstructorTests.fs (hunk 1 above)
Comment (KevinRansom) [reply tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConstructorTests.fs:29]: Hey @dsyme , indeed yes and amazingly I can even remember why ... the test verified the behaviour without preview [*Right Attribute Module Abbreviation without preview (compile)*`].  I.e. the old F#5.0 behaviour, so ... this the test became with net50 [*Right Attribute Module Abbreviation with version 5.0 (compile)*] instead

Comment (Happypig375) [reply tests/FSharp.Compiler.ComponentTests/ErrorMessages/ConstructorTests.fs:29]: This... isn't the test you are referring to.

**Pull request merged**